pkgs/tools/typesetting/tex/texlive/bin.nix

```diff
@@ -90,11 +90,12 @@ core = stdenv.mkDerivation rec {
 
   nativeBuildInputs = [
     pkg-config
-  ] ++ lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+  ] ++ lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform && self.bin.core ? __spliced) [
```

Comment:
@apfelkuchen6 this was causing infinite recursion when evaluating the splice `hostHost`, because in that case the build `self.bin.core` would be the same as the host `self.bin.core`, regardless of the true `buildPlatform` and `hostPlaftorm`. Conveniently, `__spliced` is not generated in `hostHost`, hence the `? __spliced` guard.

I don't think I have seen this pattern anywhere. Is there a more idiomatic way to address the issue?

(This is an academic issue as we would never evaluate `hostHost` in the actual build, it was just annoying for debugging.)

Reply:
There are a bunch of derivations with cross self-dependenies (for example vala and guile), but I couldn't find anything that only adds itself if the derivation is spliced.

I'd consider it a non-issue. Maybe doing nothing about it is even better as it might detect actual problems at eval time.